rita/src/rita_client/dashboard/mod.rs

```diff
@@ -156,3 +162,75 @@ impl Handler<SetWifiConfig> for Dashboard {
         Ok(())
     }
 }
+
+#[derive(Serialize)]
+pub struct NodeInfo {
+    pub nickname: String,
+    pub route_metric_to_exit: u16,
+    pub total_payments: i64,
+    pub debt: i64,
+    pub link_cost: u16,
+    pub price_to_exit: u32,
+}
+
+pub struct GetNodeInfo;
+
+impl Message for GetNodeInfo {
+    type Result = Result<Vec<NodeInfo>, Error>;
+}
+
+impl Handler<GetNodeInfo> for Dashboard {
+    type Result = ResponseFuture<Vec<NodeInfo>, Error>;
+
+    fn handle(&mut self, _msg: GetNodeInfo, _ctx: &mut Self::Context) -> Self::Result {
+        Box::new(
+            DebtKeeper::from_registry()
+                .send(Dump {})
+                .and_then(|res| {
+                    let stream = TcpStream::connect::<SocketAddr>(
+                        format!("[::1]:{}", SETTING.get_network().babel_port)
+                            .parse()
+                            .unwrap(),
+                    ).expect("Failed to connect to Babel!");
```

Comment:
bunches of expects here, mainly because the question of "where to propagate the error to?" seems to be nowhere. We can't handle this outside of the request function because the response will never reach the requester. So all errors must be resolved here.